### DISCUSSION TITLE
slurm-web using openldap but can't work

### DISCUSSION THREAD

**Discussion by emalovelife** [rackslab/Slurm-web#310] [Q&A] [answered]
i have some trouble in using your slurm-web. When i configure authentication with openldap, and i use user/password as authentication, the login web page always says " ERROR,Authentication error: Invalid user or password"
 However, i use /usr/libexec/slurm-web/slurm-web-ldap-check command, it runs ok, here's the answer
 INFO ⸬ Running slurm-web-ldap-check
 Found 2 user(s) in LDAP directory:
 - hslurm (hello_slurm) [acoustic, physic]
 - jgreen (jim_green) [acoustic, biology]

 then I'll paste some configuration in gateway.ini

 [authentication]
 enabled=yes
 method=ldap

 [ldap]
 uri=ldap://192.169.1.225
 bind_dn=cn=admin,dc=s5,dc=localdomain
 bind_password=slap123
 user_base=ou=People,dc=s5,dc=localdomain
 group_base=ou=Group,dc=s5,dc=localdomain
 user_class=posixAccount

 Through the openldap.log I knew the difference, but I don't know why the web page can't use the same procedure as the test command slurm-web-ldap-check
 Here's the slurm-web-ldap-check log message:
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 fd=13 ACCEPT from IP=172.19.129.231:55372 (IP=0.0.0.0:389)
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 op=0 BIND dn="cn=admin,dc=s5,dc=localdomain" method=128
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 op=0 BIND dn="cn=admin,dc=s5,dc=localdomain" mech=SIMPLE bind_ssf=0 ssf=0
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 op=0 RESULT tag=97 err=0 qtime=0.000054 etime=0.004250 text=
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 op=1 SRCH base="ou=People,dc=s5,dc=localdomain" scope=2 deref=0 filter="(objectClass=posixAccount)"
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 op=1 SRCH attr=uid
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 op=1 SEARCH RESULT tag=101 err=0 qtime=0.000033 etime=0.001014 nentries=2 text=
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 op=2 SRCH base="cn=hello_slurm,ou=People,dc=s5,dc=localdomain" scope=0 deref=0 filter="(objectClass=posixAccount)"
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 op=2 SRCH attr=cn gidNumber
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 op=2 SEARCH RESULT tag=101 err=0 qtime=0.000084 etime=0.000493 nentries=1 text=
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 op=3 SRCH base="ou=Group,dc=s5,dc=localdomain" scope=2 deref=0 filter="(&(|(objectClass=posixGroup)(objectClass=groupOfNames))(|(memberUid=hslurm)(member=cn=hello_slurm,ou=people,dc=s5,dc=localdomain)(gidNumber=500)))"
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 op=3 SRCH attr=cn
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 op=3 SEARCH RESULT tag=101 err=0 qtime=0.000067 etime=0.000414 nentries=2 text=
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 op=4 SRCH base="cn=jim_green,ou=People,dc=s5,dc=localdomain" scope=0 deref=0 filter="(objectClass=posixAccount)"
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 op=4 SRCH attr=cn gidNumber
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 op=4 SEARCH RESULT tag=101 err=0 qtime=0.000028 etime=0.000208 nentries=1 text=
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 op=5 SRCH base="ou=Group,dc=s5,dc=localdomain" scope=2 deref=0 filter="(&(|(objectClass=posixGroup)(objectClass=groupOfNames))(|(memberUid=jgreen)(member=cn=jim_green,ou=people,dc=s5,dc=localdomain)(gidNumber=501)))"
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 op=5 SRCH attr=cn
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 op=5 SEARCH RESULT tag=101 err=0 qtime=0.000019 etime=0.000277 nentries=2 text=
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 op=6 UNBIND
 Jun 21 15:27:48 s5 slapd[534541]: conn=1049 fd=13 closed

 While the login web page has the log message:
 Jun 21 15:29:14 s5 slapd[534541]: conn=1050 fd=13 ACCEPT from IP=172.19.129.231:60966 (IP=0.0.0.0:389)
 Jun 21 15:29:14 s5 slapd[534541]: conn=1050 op=0 BIND dn="uid=jgreen,ou=People,dc=s5,dc=localdomain" method=128
 Jun 21 15:29:14 s5 slapd[534541]: conn=1050 op=0 RESULT tag=97 err=49 qtime=0.000055 etime=0.000669 text=
 Jun 21 15:29:14 s5 slapd[534541]: conn=1050 op=1 UNBIND
 Jun 21 15:29:14 s5 slapd[534541]: conn=1050 fd=13 closed

 Obviously, the web login has less procedure than the test command, it didn't use the data that i configure in the gateway.ini such as "bind_dn=cn=admin,dc=s5,dc=localdomain
 bind_password=slap123", i don't know why

i even tried
user_fullname_attribute=cn
group_name_attribute=cn

and tried fullname in login box, still can't work

here's my openldap ldif dumpfile
##
## Version: 1.2.6.6

version: 1

##Entry 1: dc=s5,dc=localdomain
dn: dc=s5,dc=localdomain
dc: s5
objectclass: top
objectclass: domain

##Entry 2: cn=admin,dc=s5,dc=localdomain
dn: cn=admin,dc=s5,dc=localdomain
cn: admin
description: LDAP Manager
objectclass: organizationalRole

##Entry 3: ou=Group,dc=s5,dc=localdomain
dn: ou=Group,dc=s5,dc=localdomain
objectclass: organizationalUnit
ou: Group

##Entry 4: cn=acoustic,ou=Group,dc=s5,dc=localdomain
dn: cn=acoustic,ou=Group,dc=s5,dc=localdomain
cn: acoustic
gidnumber: 501
memberuid: hslurm
objectclass: posixGroup
objectclass: top

##Entry 5: cn=biology,ou=Group,dc=s5,dc=localdomain
dn: cn=biology,ou=Group,dc=s5,dc=localdomain
cn: biology
gidnumber: 503
memberuid: jgreen
objectclass: posixGroup
objectclass: top

##Entry 6: cn=physic,ou=Group,dc=s5,dc=localdomain
dn: cn=physic,ou=Group,dc=s5,dc=localdomain
cn: physic
gidnumber: 502
memberuid: hslurm
objectclass: posixGroup
objectclass: top

##Entry 7: ou=People,dc=s5,dc=localdomain
dn: ou=People,dc=s5,dc=localdomain
objectclass: organizationalUnit
ou: People

##Entry 8: cn=hello_slurm,ou=People,dc=s5,dc=localdomain
dn: cn=hello_slurm,ou=People,dc=s5,dc=localdomain
cn: hello_slurm
gidnumber: 500
givenname: hello
homedirectory: /home/users/hslurm
loginshell: /usr/sbin/nologin
objectclass: inetOrgPerson
objectclass: posixAccount
objectclass: top
sn: slurm
uid: hslurm
uidnumber: 1001
userpassword: 123456

##Entry 9: cn=jim_green,ou=People,dc=s5,dc=localdomain
dn: cn=jim_green,ou=People,dc=s5,dc=localdomain
cn: jim_green
gidnumber: 501
givenname: jim
homedirectory: /home/users/jgreen
loginshell: /usr/sbin/nologin
objectclass: inetOrgPerson
objectclass: posixAccount
objectclass: top
sn: Green
uid: jgreen
uidnumber: 1002
userpassword: 123456

**Comment by emalovelife** [comment]
Here's the author's reply, but i still can't fixed it,  i doubt the web login code don't work as the test command well, is anybody tried this, can others give me a detailed configuration

Thank you for contacting us, I'm glad you are interested in Slurm-web and you started testing!

Do you have a Github account? If yes, I invite you to ask technical questions on the community channel at this URL:
https://github.com/rackslab/Slurm-web/discussions/categories/q-a

This way, other people will benefit from the answers.

Your sldapd logs indicates:

> Jun 21 15:29:14 s5 slapd[534541]: conn=1050 op=0 RESULT tag=97 err=49 qtime=0.000055 etime=0.000669 text=
                                                                 ^^^^^^
In OpenLDAP, error 49 means "Invalid credentials". This either means the user dn "uid=jgreen,ou=People,dc=s5,dc=localdomain" or the password is wrong. Do you know what is the dn of this user entry in your LDAP directory?

For more useful information about authentication logic in Slurm-web logs, you can enable debug messages in gateway.ini:

[service]
debug=yes
debug_flags=
  slurmweb
  rfl

Then restart the gateway service and checkout the logs.

You can also find more information about Slurm-web logs in troubleshooting guide: https://docs.rackslab.io/slurm-web/misc/troubleshooting.html

Best regards,
Rémi

**Comment by rezib** [comment, marked as answer]
Hello @emalovelife,

The dn of the user entries in your LDAP directory use the cn attribute. This is not yet supported in Slurm-web 3.0.0, this limitation was reported in https://github.com/rackslab/Slurm-web/issues/305. A fix has been developed and integrated, it will be released with Slurm-web 3.1.0 (probably next week).

Then, in order to support your LDAP tree, you will need to upgrade to Slurm-web 3.1.0 and set this parameter in your `gateway.ini`:

```
[ldap]
user_name_attribute = cn
```

Then you will be able to login with logins _jim_green_ or _hello_slurm_ and their passwords.

**Reply by rezib** [reply]
Slurm-web v3.1.0 is now available: https://github.com/rackslab/Slurm-web/discussions/311

@emalovelife, can you try the suggestion above?

**Reply by emalovelife** [reply]
Yes, i did. and it worked well.
I tried 3.1 source code installation, the user authentication passed, but i can't see the main page. After I strived hard to make 3.1 rpm package and installed them, i really got the right results
here's my part of configuration
gateway.ini

`
[authentication]
enabled=yes
method=ldap

[ldap]
uri=ldap://192.169.1.225

bind_dn=cn=system_admin,ou=People,dc=s5,dc=localdomain

bind_password=654321

user_base=ou=People,dc=s5,dc=localdomain
group_base=ou=Group,dc=s5,dc=localdomain

user_name_attribute=cn
user_fullname_attribute=cn
group_name_attribute=cn
user_class=posixAccount

restricted_groups=
  admins
  biology

`
agent.ini
`[policy]

definition=/usr/share/slurm-web/conf/policy.yml
vendor_roles=/usr/share/slurm-web/conf/policy.ini
roles=/etc/slurm-web/policy.ini
`

policy.ini
`[roles]
# Enable anonymous role with basic views
anonymous
# All authenticated users have the user role
user=ALL
# Admin role for it group members
admin=@admins

[anonymous]
actions=view-stats

[user]
actions=view-stats,view-jobs,view-accounts

[admin]
actions=view-nodes,view-partitions,view-qos,view-reservations
`


and My ldap layout like this
`
dc=s5,dc=localdomain
|
|--ou=Group
|    |-- cn=acoustic
|    |-- cn=admins
|    |-- cn=biology 
|--ou=People
     |-- cn=hello_slurm (in biology)
     |-- cn=jim_green   (in acoustic)
     |-- cn=system_admin(in admins)

`

when i login hell_slurm, it show part of the view rights, when i login jim_green it shows it's not in the restricted groups, when I login system_admin, it has the most view rights.
Thanks all of your developing work

**Reply by luckask3** [reply]
Hello @emalovelife 

I have some doubts when configuring phpldapadmin to authenticate with slurm-web, when you log in with the users you created, do you do a full login like in this example "cn=system_admin,ou=People,dc=s5,dc=localdomain" or do you just enter the login "hello_slurm" and your password?

**Reply by luckask3** [reply]
I have the same problem you described, I have the users but I can't log in to slurm-web:

INFO ⸬ Running slurm-web-ldap-check
Found 1 user(s) in LDAP directory:
- lucas.borges (Lucas Borges) [adminuser, groups]

If you can attach a screenshot of your ldap configuration I would be grateful.

**Reply by rezib** [reply]
@luckask3 Can you copy/paste your user entry as reported by `ldapsearch`? It will give most information to help you define correct LDAP settings.

Can you also give an extract of gateway service logs upon a failed login attempt?

**Reply by emalovelife** [reply]
> Hello @emalovelife
> 
> I have some doubts when configuring phpldapadmin to authenticate with slurm-web, when you log in with the users you created, do you do a full login like in this example "cn=system_admin,ou=People,dc=s5,dc=localdomain" or do you just enter the login "hello_slurm" and your password?

in web page, there're two edit boxes, and i use "system_admin" as the user,"654321" as the password, then i tried "jim_green" as the user, "123456" as the password.  it can access the main page. and the layout of my phpldapadmin is below. 
make sure you use 3.1 versions and make sure your gateway.ini like my example(ignore the first config in the phpldap config, see the config layout below). if you have some trouble, you can see the openldap.log if you configure the logfile, or you can install wireshark and see the the real content in pcap package

**Reply by luckask3** [reply]
@rezib This is my config ldapsearch:

```ldif
# extended LDIF
#
# LDAPv3
# base <dc=dpihpc,dc=mao,dc=org,dc=br> with scope subtree
# filter: (objectclass=*)
# requesting: ALL
#

# dpihpc.mao.org.br
dn: dc=dpihpc,dc=mao,dc=org,dc=br
objectClass: top
objectClass: dcObject
objectClass: organization
o: HPC DPI
dc: dpihpc

# admin, dpihpc.mao.org.br
dn: cn=admin,dc=dpihpc,dc=mao,dc=org,dc=br
objectClass: simpleSecurityObject
objectClass: organizationalRole
cn: admin
description: LDAP administrator
userPassword:: e1NTSEF9bjNMVzdndWdRQzY2V1hTcDJrdjhuMUkwNDhUcDdaUUw=

# group, dpihpc.mao.org.br
dn: ou=group,dc=dpihpc,dc=mao,dc=org,dc=br
ou: group
objectClass: organizationalUnit
objectClass: top

# people, dpihpc.mao.org.br
dn: ou=people,dc=dpihpc,dc=mao,dc=org,dc=br
ou: people
objectClass: organizationalUnit
objectClass: top

# admins, group, dpihpc.mao.org.br
dn: cn=admins,ou=group,dc=dpihpc,dc=mao,dc=org,dc=br
cn: admins
gidNumber: 500
objectClass: posixGroup
objectClass: top

# others, group, dpihpc.mao.org.br
dn: cn=others,ou=group,dc=dpihpc,dc=mao,dc=org,dc=br
cn: others
gidNumber: 502
objectClass: posixGroup
objectClass: top

# biology, group, dpihpc.mao.org.br
dn: cn=biology,ou=group,dc=dpihpc,dc=mao,dc=org,dc=br
cn: biology
gidNumber: 501
objectClass: posixGroup
objectClass: top

# test slurm, people, dpihpc.mao.org.br
dn: cn=test slurm,ou=people,dc=dpihpc,dc=mao,dc=org,dc=br
givenName: test
sn: slurm
cn: test slurm
uid: test_slurm
userPassword:: e01ENX12Z204amFvOWVPTkNYVFpHTTVYdUhnPT0=
uidNumber: 1001
gidNumber: 501
homeDirectory: /home/users/test_slurm
objectClass: inetOrgPerson
objectClass: posixAccount
objectClass: top

# test others, people, dpihpc.mao.org.br
dn: cn=test others,ou=people,dc=dpihpc,dc=mao,dc=org,dc=br
givenName: test
sn: others
cn: test others
uid: test_others
userPassword:: e01ENX12Z204amFvOWVPTkNYVFpHTTVYdUhnPT0=
uidNumber: 1002
gidNumber: 502
homeDirectory: /home/users/test_others
objectClass: inetOrgPerson
objectClass: posixAccount
objectClass: top

# lucas borges, people, dpihpc.mao.org.br
dn: cn=lucas borges,ou=people,dc=dpihpc,dc=mao,dc=org,dc=br
givenName: lucas
sn: borges
cn: lucas borges
uid: lucas.borges
userPassword:: e01ENX12Z204amFvOWVPTkNYVFpHTTVYdUhnPT0=
uidNumber: 1000
gidNumber: 500
homeDirectory: /home/users/lucas.borges
objectClass: inetOrgPerson
objectClass: posixAccount
objectClass: top

# search result
search: 2
result: 0 Success

# numResponses: 11
# numEntries: 10
```

EDIT: markdown formatting to improve readability of LDIF output.

**Reply by luckask3** [reply]
/usr/libexec/slurm-web/slurm-web-ldap-check
INFO ⸬ Running slurm-web-ldap-check
Found 3 user(s) in LDAP directory:
- lucas.borges (lucas borges) [admins]
- test_slurm (test slurm) [biology]
- test_others (test others) [others]

**Reply by luckask3** [reply]
And my configuration gateway.ini:

```ini
[authentication]

# Determine if authentication is enabled
enabled=yes

# Authentification method

method=ldap

[ldap]

# URI to connect to LDAP server
uri=ldap://192.168.0.1:389

# default system CA certificates is used.
cacert=/path/to/certificate.pem

# Use STARTTLS protocol to negociate TLS connection with LDAP server
starttls=no

# Base DN for users entries
#user_base=ou=people,dc=example,dc=org
user_base=ou=people,dc=dpihpc,dc=mao,dc=org,dc=br

# Base DN for group entries
#group_base=ou=group,dc=example,dc=org
group_base=ou=group,dc=dpihpc,dc=mao,dc=org,dc=br

# Class of user entries
#
# Default value: posixAccount
user_class=posixAccount

# User entry attribute for user name
# Default value: uid
user_name_attribute=cn

# User entry attribute for full name
# Default value: cn
user_fullname_attribute=cn

# Group entry attribute for name
# Default value: cn
group_name_attribute=cn

# DN used to bind to the LDAP server. When this parameter is not defined,
# access to LDAP directory is performed anonymously.
#bind_dn=cn=system,ou=people,dc=example,dc=org
bind_dn=cn=admin,dc=dpihpc,dc=mao,dc=org,dc=br

# Password of bind DN. This parameter is required when `bind_dn` is
# defined
bind_password=bjj2020

# List of users groups allowed to connect. When this parameter is not
# defined, all users in LDAP directory are authorized to sign in.
restricted_groups=
  admins
  biology
  others
```

EDIT: markdown formatting to improve readability on config file.

**Reply by luckask3** [reply]
When I try make login on Slurm-web I received this message error:

Authentication error: Unable to find group base ou=group,dc=dpihpc,dc=mao,dc=org,dc=br

**Reply by rezib** [reply]
Well, I don't see anything wrong in your settings. Maybe debug logs could help.

Can you please add the following settings in your `gateway.ini`:

```ìni
[service]
debug=yes
debug_flags=
  slurmweb
  rfl
```

Then restart `slurm-web-gateway.service`, try to login (unsucessfully), and send us logs extracted from `journalctl -u slurm-web-gateway.service` output.

**Reply by luckask3** [reply]
@rezib output journalctl:

```
Aug 05 13:41:47 hpc-server slurm-web-gateway[110797]: MainThread: [DEBUG] Loading JWT private key from path /var/lib/slurm-web/jwt.key
Aug 05 13:41:47 hpc-server slurm-web-gateway[110797]: MainThread: [INFO] Running slurm-web-gateway application
Aug 05 13:41:47 hpc-server slurm-web-gateway[110797]: * Serving Flask app 'slurm-web-gateway'
Aug 05 13:41:47 hpc-server slurm-web-gateway[110797]: * Debug mode: off
Aug 05 13:41:47 hpc-server slurm-web-gateway[110797]: MainThread: [INFO] WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
Aug 05 13:41:47 hpc-server slurm-web-gateway[110797]: * Running on http://192.168.0.1:5011/
Aug 05 13:41:47 hpc-server slurm-web-gateway[110797]: MainThread: [INFO] Press CTRL+C to quit
Aug 05 13:42:00 hpc-server systemd[1]: Stopping slurm-web-gateway.service - Slurm-web HPC dashboard gateway...
Aug 05 13:42:00 hpc-server systemd[1]: slurm-web-gateway.service: Deactivated successfully.
Aug 05 13:42:00 hpc-server systemd[1]: Stopped slurm-web-gateway.service - Slurm-web HPC dashboard gateway.
Aug 05 13:42:00 hpc-server systemd[1]: Started slurm-web-gateway.service - Slurm-web HPC dashboard gateway.
Aug 05 13:42:00 hpc-server slurm-web-gateway[110819]: MainThread: [DEBUG] Loading JWT private key from path /var/lib/slurm-web/jwt.key
Aug 05 13:42:00 hpc-server slurm-web-gateway[110819]: MainThread: [INFO] Running slurm-web-gateway application
Aug 05 13:42:00 hpc-server slurm-web-gateway[110819]: * Serving Flask app 'slurm-web-gateway'
Aug 05 13:42:00 hpc-server slurm-web-gateway[110819]: * Debug mode: off
Aug 05 13:42:00 hpc-server slurm-web-gateway[110819]: MainThread: [INFO] WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
Aug 05 13:42:00 hpc-server slurm-web-gateway[110819]: * Running on http://192.168.0.1:5011/
Aug 05 13:42:00 hpc-server slurm-web-gateway[110819]: MainThread: [INFO] Press CTRL+C to quit
Aug 05 13:42:05 hpc-server slurm-web-gateway[110819]: Thread-1 (process_request_thread): [DEBUG] LDAP search base: cn=lucas.borges,ou=people,dc=dpihpc,dc=mao,dc=org,dc=br, scope: base filter: (objectClass=posixAccount), results: [('cn=lucas.borges,ou=people,dc=dpihpc,dc=mao,dc=org,dc=br', {'gidNumber': [b'500'], 'cn': [b'lucas.borges']})]
Aug 05 13:42:05 hpc-server slurm-web-gateway[110819]: Thread-1 (process_request_thread): [WARNING] LDAP authentication error for user lucas.borges: Unable to find group base ou=group,dc=dpihpc,dc=mao,dc=org,dc=br
Aug 05 13:42:05 hpc-server slurm-web-gateway[110819]: Thread-1 (process_request_thread): [INFO] 8.8.8.8 - - [05/Aug/2024 13:42:05] "POST /api/login HTTP/1.1" 401 -
```

EDIT: markdown format of logs.

**Reply by rezib** [reply]
I find it weird the request for `cn=lucas.borges,ou=people,dc=dpihpc,dc=mao,dc=org,dc=br` has a result while the LDIF above has `cn=lucas borges` (with a whitespace instead of a dot). Can you please send the output of this command:

```sh
# /usr/libexec/slurm-web/slurm-web-ldap-check --debug --debug-flags rfl
```

**Reply by luckask3** [reply]
```
@rezib 
[DEBUG] ⸬rfl.settings.loaders:30        ↦ Loading site settings file /etc/slurm-web/gateway.ini
[INFO]  ⸬slurmweb.apps.ldap:22          ↦ Running slurm-web-ldap-check
[DEBUG] ⸬rfl.authentication.ldap:283    ↦ Using DN cn=admin,dc=dpihpc,dc=mao,dc=org,dc=br to bind to LDAP directory
[DEBUG] ⸬rfl.authentication.ldap:252    ↦ LDAP search base: ou=group,dc=dpihpc,dc=mao,dc=org,dc=br, scope: subtree, filter: (objectClass=posixAccount), results: [('cn=lucas.borges,ou=people,dc=dpihpc,dc=mao,dc=org,dc=br', {'uid': [b'lucas.borges']}), ('cn=test_slurm,ou=people,dc=dpihpc,dc=mao,dc=org,dc=br', {'uid': [b'test_slurm']}), ('cn=test_others,ou=people,dc=dpihpc,dc=mao,dc=org,dc=br', {'uid': [b'test_others']})]
[DEBUG] ⸬rfl.authentication.ldap:111    ↦ LDAP search base: cn=lucas.borges,ou=people,dc=dpihpc,dc=mao,dc=org,dc=br, scope: base filter: (objectClass=posixAccount), results: [('cn=lucas.borges,ou=people,dc=dpihpc,dc=mao,dc=org,dc=br', {'gidNumber': [b'500'], 'cn': [b'lucas.borges']})]
[DEBUG] ⸬rfl.authentication.ldap:173    ↦ LDAP search base: ou=group,dc=dpihpc,dc=mao,dc=org,dc=br, scope: subtree, filter: (&(|(objectClass=posixGroup)(objectClass=groupOfNames))(|(memberUid=lucas.borges)(member=cn=lucas.borges,ou=people,dc=dpihpc,dc=mao,dc=org,dc=br)(gidNumber=500))), results: [('cn=admins,ou=group,dc=dpihpc,dc=mao,dc=org,dc=br', {'cn': [b'admins']})]
[DEBUG] ⸬rfl.authentication.ldap:111    ↦ LDAP search base: cn=test_slurm,ou=people,dc=dpihpc,dc=mao,dc=org,dc=br, scope: base filter: (objectClass=posixAccount), results: [('cn=test_slurm,ou=people,dc=dpihpc,dc=mao,dc=org,dc=br', {'gidNumber': [b'501'], 'cn': [b'test_slurm']})]
[DEBUG] ⸬rfl.authentication.ldap:173    ↦ LDAP search base: ou=group,dc=dpihpc,dc=mao,dc=org,dc=br, scope: subtree, filter: (&(|(objectClass=posixGroup)(objectClass=groupOfNames))(|(memberUid=test_slurm)(member=cn=test_slurm,ou=people,dc=dpihpc,dc=mao,dc=org,dc=br)(gidNumber=501))), results: [('cn=biology,ou=group,dc=dpihpc,dc=mao,dc=org,dc=br', {'cn': [b'biology']})]
[DEBUG] ⸬rfl.authentication.ldap:111    ↦ LDAP search base: cn=test_others,ou=people,dc=dpihpc,dc=mao,dc=org,dc=br, scope: base filter: (objectClass=posixAccount), results: [('cn=test_others,ou=people,dc=dpihpc,dc=mao,dc=org,dc=br', {'gidNumber': [b'502'], 'cn': [b'test_others']})]
[DEBUG] ⸬rfl.authentication.ldap:173    ↦ LDAP search base: ou=group,dc=dpihpc,dc=mao,dc=org,dc=br, scope: subtree, filter: (&(|(objectClass=posixGroup)(objectClass=groupOfNames))(|(memberUid=test_others)(member=cn=test_others,ou=people,dc=dpihpc,dc=mao,dc=org,dc=br)(gidNumber=502))), results: [('cn=others,ou=group,dc=dpihpc,dc=mao,dc=org,dc=br', {'cn': [b'others']})]
Found 3 user(s) in LDAP directory:
- lucas.borges (lucas.borges) [admins]
- test_slurm (test_slurm) [biology]
- test_others (test_others) [others]
```

**Reply by luckask3** [reply]
I'm using phpldapadmin version 1.2.5 in a docker container, do you think I should upgrade to a higher version?

**Reply by rezib** [reply]
OK thank you for these inputs. I think the problem is due to your users (such as `lucas.borges`) not being permitted to access the `ou=group,dc=dpihpc,dc=mao,dc=org,dc=br` subtree. It looks like the dn `cn=admin,dc=dpihpc,dc=mao,dc=org,dc=br` has access to this subtree but not other users. You should verify this point.

**Reply by luckask3** [reply]
I still haven't quit very well how phpldap's directory and subdirectory policy works.

By default, there is a cn=admin already created, within this default cn I created the ou=group with its respective groups and the ou=people with its respective users pointing to their groups. Is this structure correct?


```
+--> dc=dpihpc,dc=mao,dc=org,dc=br (1)
  +--> cn=admin (2)
  | +--> ou=group (3)
  | | ---> cn=admins
  | | ---> cn=biology
  | | ---> cn=others
  | | ---> Create new entry here
  | +--> ou=people (3)
  | | ---> cn=lucas.borges
  | | ---> cn=test.others
  | | ---> cn=test.slurm
```

**Reply by rezib** [reply]
The structure is OK, it is all about the permissions to read the tree for all users. You should probably have a look at: https://www.openldap.org/doc/admin24/access-control.html

**Reply by luckask3** [reply]
Hello @rezib,

Thanks for your support, this issue has been resolved. The docker container for some reason was causing authentication to fail, I configured phpldap directly on linux and was successful with the configuration.

**Reply by luckask3** [reply]
@rezib  I have a question.
Is possible to run Jobs through the slurm-web interface? I did not find this functionality or any related theme in the documentation.

**Reply by rezib** [reply]
> Is possible to run Jobs through the slurm-web interface? I did not find this functionality or any related theme in the documentation.

It is in the roadmap: https://github.com/rackslab/Slurm-web/issues/281

Rackslab is looking for sponsor to fund the development of this feature, would your organization like to participate?

**Reply by luckask3** [reply]
I wish that were possible, but I can't answer that for you.

I am a little new to this, so please pardon my ignorance.
I have some doubts regarding user authentication to my slurm cluster to run jobs, since it is not possible to run jobs through slurm-web I will have to find another way to perform authentication.
The documentation is not very clear about using LDAP accounts.
Saw somewhere in the list about using UsePAM=1 and copying or creating a softlink for slurm PAM module under /etc/pam.d . But it didn't work for me.
I am bereft of ideas at present. My question is how is it possible to perform authentication so that a user can perform tasks in my cluster trough ssh. Do you have any tutorial or documentation that can help me with this?

**Reply by rezib** [reply]
I am sorry but I don't do Slurm support in this channel. If you need help to setup a cluster with Slurm, SSH and LDAP, [Rackslab](https://rackslab.io) offers professional services in HPC, feel free to [contact us](https://rackslab.io/en/contact/) if you want a quotation.

**Reply by emalovelife** [reply]
> I still haven't quit very well how phpldap's directory and subdirectory policy works.
> 
> By default, there is a cn=admin already created, within this default cn I created the ou=group with its respective groups and the ou=people with its respective users pointing to their groups. Is this structure correct?
> 
> ```
> +--> dc=dpihpc,dc=mao,dc=org,dc=br (1)
>   +--> cn=admin (2)
>   | +--> ou=group (3)
>   | | ---> cn=admins
>   | | ---> cn=biology
>   | | ---> cn=others
>   | | ---> Create new entry here
>   | +--> ou=people (3)
>   | | ---> cn=lucas.borges
>   | | ---> cn=test.others
>   | | ---> cn=test.slurm
> ```

seems in your ldif file, People and Group are the same level with admin, however your layout picture show that People and Group are under  admin.  while in my configuration, they are in the same level.

**Comment by emalovelife** [comment]
in fact it's like this, the users are under People, the editor does not show what i mean
`
dc=s5,dc=localdomain
|
|--ou=Group
|    |-- cn=acoustic
|    |-- cn=admins
|    |-- cn=biology 
|--ou=People
|    |-- cn=hello_slurm (in biology)
|    |-- cn=jim_green   (in acoustic)
|    |-- cn=system_admin(in admins)

`

**Reply by luckask3** [reply]
Thank you @emalovelife I applied a similar structure to my settings.
